import/popups/AuthDialog.qml

```diff
@@ -60,23 +68,36 @@ UserPrompt {
             width: parent.width
             //% "Enter password"
             placeholderText: qsTrId("sailfish_components_webview_popups-la-enter_password")
+            text: root.password && root.password.value || ""
+
             EnterKey.iconSource: (username.text.length > 0 && text.length > 0) ? "image://theme/icon-m-enter-accept"
                                                                                : "image://theme/icon-m-enter-next"
-            EnterKey.onClicked: dialog.accept()
+            EnterKey.onClicked: root.accept()
         }
 
         TextSwitch {
-            id: dontsaveCheck
+            id: remember
 
+            // If credentials are already remember removing them via this is not feasibible
+            // Better to hide the whole checkbox.
+            visible: !privateBrowsing && !(root.remember && root.remember.checked)
             checked: false
 
-            //: Do not save the entered credentials for later use
-            //% "Do not save"
-            text: qsTrId("sailfish_components_webview_popups-la-dont_save")
+            //: Remember entered credentials for later use
+            //% "Remember credentials"
+            text: qsTrId("sailfish_components_webview_popups-remember_credentials")
+        }
+
+        Label {
+            x: Theme.paddingLarge
+            width: parent.width - Theme.paddingLarge * 2
+            visible: privateBrowsing
+            wrapMode: Text.Wrap
+            color: Theme.highlightColor
 
-            //: Explain to the user that checking the switch will prevent entered credentials from saving
-            //% "The credentials you enter won't be stored for later use"
-            description: qsTrId("sailfish_components_webview_popups-la-dont_save_description")
+            //: Description label for user when private mode active (entered credentials are not saved)
+            //% "The entered credentials won't never be stored in private browsing mode"
```

Comment:
It should be "won't ever", but I suggest a drier:
"Credentials cannot be stored in private browsing mode"

Reply:
Technically we probably could  store then. Hence, I'd put more emphasis to the behaviour. Should be it like:
"Credential are not stored in private browsing mode"

Reply:
The 'cannot' is more of a 'this program does not make it possible' sort of thing. 'Are not' would suggest that the storage has no effect. Could go with an unambigious 

'Credential storage is disabled in private browsing mode'

Reply:
Yeah that sounds better. Maybe "not disabled"... but "not available"? How does that sound like to you?

Reply:
Yes, that's better. 'Disabled' might make people think they can re-enable it in some setting somewhere.

Reply:
Now amended. Thanks @abranson for this ping-pong and formatting this one.